src/Storages/StorageMerge.cpp

```diff
@@ -267,7 +265,7 @@ Pipe StorageMerge::read(
     auto pipe = Pipe::unitePipes(std::move(pipes));
 
     if (!pipe.empty())
-        narrowPipe(pipe, num_streams);
```

Comment:
`narrowPipe` had mush more complicated logic than `resize`.
It doesn't look like just refactoring.

Reply:
Yeah. I'll resurrect it and add comments instead.